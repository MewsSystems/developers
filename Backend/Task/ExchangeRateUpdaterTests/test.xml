<?xml version="1.0" encoding="UTF-8"?>
<kurzy banka="CNB" datum="16.01.2020" poradi="11">
<tabulka typ="XML_TYP_CNB_KURZY_DEVIZOVEHO_TRHU">
<radek kod="AUD" mena="dolar" mnozstvi="1" kurz="15,622" zeme="Austrálie"/>
<radek kod="BRL" mena="real" mnozstvi="1" kurz="5,411" zeme="Brazílie"/>
<radek kod="BGN" mena="lev" mnozstvi="1" kurz="12,870" zeme="Bulharsko"/>
<radek kod="CNY" mena="žen-min-pi" mnozstvi="1" kurz="3,276" zeme="Čína"/>
<radek kod="DKK" mena="koruna" mnozstvi="1" kurz="3,368" zeme="Dánsko"/>
<radek kod="EUR" mena="euro" mnozstvi="1" kurz="25,170" zeme="EMU"/>
<radek kod="PHP" mena="peso" mnozstvi="100" kurz="44,330" zeme="Filipíny"/>
<radek kod="HKD" mena="dolar" mnozstvi="1" kurz="2,900" zeme="Hongkong"/>
<radek kod="HRK" mena="kuna" mnozstvi="1" kurz="3,380" zeme="Chorvatsko"/>
<radek kod="INR" mena="rupie" mnozstvi="100" kurz="31,779" zeme="Indie"/>
<radek kod="IDR" mena="rupie" mnozstvi="1000" kurz="1,654" zeme="Indonesie"/>
<radek kod="ISK" mena="koruna" mnozstvi="100" kurz="18,372" zeme="Island"/>
<radek kod="ILS" mena="nový šekel" mnozstvi="1" kurz="6,526" zeme="Izrael"/>
<radek kod="JPY" mena="jen" mnozstvi="100" kurz="20,497" zeme="Japonsko"/>
<radek kod="ZAR" mena="rand" mnozstvi="1" kurz="1,562" zeme="Jižní Afrika"/>
<radek kod="CAD" mena="dolar" mnozstvi="1" kurz="17,294" zeme="Kanada"/>
<radek kod="KRW" mena="won" mnozstvi="100" kurz="1,945" zeme="Korejská republika"/>
<radek kod="HUF" mena="forint" mnozstvi="100" kurz="7,540" zeme="Maďarsko"/>
<radek kod="MYR" mena="ringgit" mnozstvi="1" kurz="5,546" zeme="Malajsie"/>
<radek kod="MXN" mena="peso" mnozstvi="1" kurz="1,199" zeme="Mexiko"/>
<radek kod="XDR" mena="ZPČ" mnozstvi="1" kurz="31,135" zeme="MMF"/>
<radek kod="NOK" mena="koruna" mnozstvi="1" kurz="2,541" zeme="Norsko"/>
<radek kod="NZD" mena="dolar" mnozstvi="1" kurz="15,012" zeme="Nový Zéland"/>
<radek kod="PLN" mena="zlotý" mnozstvi="1" kurz="5,947" zeme="Polsko"/>
<radek kod="RON" mena="leu" mnozstvi="1" kurz="5,266" zeme="Rumunsko"/>
<radek kod="RUB" mena="rubl" mnozstvi="100" kurz="36,587" zeme="Rusko"/>
<radek kod="SGD" mena="dolar" mnozstvi="1" kurz="16,753" zeme="Singapur"/>
<radek kod="SEK" mena="koruna" mnozstvi="1" kurz="2,382" zeme="Švédsko"/>
<radek kod="CHF" mena="frank" mnozstvi="1" kurz="23,440" zeme="Švýcarsko"/>
<radek kod="THB" mena="baht" mnozstvi="100" kurz="74,222" zeme="Thajsko"/>
<radek kod="TRY" mena="lira" mnozstvi="1" kurz="3,856" zeme="Turecko"/>
<radek kod="USD" mena="dolar" mnozstvi="1" kurz="22,538" zeme="USA"/>
<radek kod="GBP" mena="libra" mnozstvi="1" kurz="29,448" zeme="Velká Británie"/>
</tabulka>
</kurzy>
