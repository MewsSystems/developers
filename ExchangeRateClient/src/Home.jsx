// @flow
import React, { Component } from 'react';
import { List } from 'immutable';
import { connect } from 'react-redux';
import styled from 'styled-components';

import type { CurrencyPairRate } from './types';
import type { RequestStatus, CurrencyPair } from './reducer';
import { getCurrencyPairList, getCurrencyPairRates } from './selectors';
import { fetchConfig, fetchRates } from './actions';
import CurrencyPairsFilter from './CurrencyPairsFilter';
import RateList from './RateList';

type Props = {
  configRequestStatus: RequestStatus,
  currencyPairs: List<CurrencyPair>,
  rates: List<CurrencyPairRate>,
  fetchConfig: () => void,
  fetchRates: (interval: number, ids: Array<string>) => void,
};

type State = {};

const Container = styled.div`
  height: 100vh;
  width: 100vw;
  padding: 1rem 2rem;
`;

const Heading = styled.h1`
  margin: 0 0 2rem;
  font-size: 2rem;
`;

const StyledFilter = styled(CurrencyPairsFilter)`
  margin: 0 0 1rem;
`;

const StyledRateList = styled(RateList)`
  margin: 0 0 1rem;
`;

class Home extends Component<Props, State> {
  static defaultProps = {};

  state = {};

  componentDidMount() {
    this.props.fetchConfig();
  }

  render() {
    const { configRequestStatus, currencyPairs, rates } = this.props;

    return (
      <Container>
        <Heading>Exchange Rate App</Heading>
        {configRequestStatus === 'success' ? (
          <>
            <StyledFilter currencyPairs={currencyPairs} />
            <StyledRateList rates={rates} />
          </>
        ) : (
          <div>Loading...</div>
        )}
      </Container>
    );
  }
}

const mapStateToProps = state => ({
  configRequestStatus: state.getIn(['app', 'configRequestStatus']),
  currencyPairs: getCurrencyPairList(state),
  rates: getCurrencyPairRates(state),
});

const mapDispatchToProps = { fetchConfig, fetchRates };

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Home);
