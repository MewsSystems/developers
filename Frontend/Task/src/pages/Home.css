#root {
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  font-family: Arial, Helvetica, sans-serif;
  font-size: calc(1vw + 1vh + 0.5vmin);
}

body {
  background-color: black;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  font-family: Arial, Helvetica, sans-serif;
}

.app__container {
  margin: 0;
  padding: 0;
  display: grid;
  grid-auto-flow: row;
  place-items: center;
  justify-content: space-around;
  align-content: start;
  width: 100vw;
  min-height: 100vh;
  overflow-x: hidden;
  background-color: darkgrey;
}
