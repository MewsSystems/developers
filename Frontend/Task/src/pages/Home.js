import React from "react";
import SelectPanel from "../components/SelectPanel";
import CurrencyDisplay from "../components/CurrencyDisplay";

import styles from "./Home.css";

const Home = props => {
  return (
    <div className={styles.app__container}>
      <SelectPanel />
      <CurrencyDisplay />
    </div>
  );
};

export default Home;
