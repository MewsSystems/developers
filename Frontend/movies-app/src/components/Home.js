import React, { useState, useEffect } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import URLSearchParams from 'query-string';

import * as actions from './../data/state/actions';
import useDebounce from './../hooks/useDebounce';
import MovieList from './MovieList';
import SearchBox from './SearchBox';
import Pagination from './Pagination';
import { Section, Row, Paragraph } from './styled';

const Home = () => {
  const { search } = useLocation();
  const params = URLSearchParams.parse(search);
  const pageIndex = params['page'] || 1;

  const history = useHistory();

  const dispatch = useDispatch();
  const movies = useSelector(state => state.movies.items);
  const currentPage = useSelector(state => state.movies.currentPage);
  const totalPages = useSelector(state => state.movies.totalPages);
  const [searchTerm, setSearchTerm] = useState('');
  const debouncedSearchTerm = useDebounce(searchTerm, 500);

  useEffect(() => {
    dispatch(actions.LoadMovies(debouncedSearchTerm, pageIndex));
  }, [debouncedSearchTerm, dispatch, pageIndex]);

  const pageChanged = newPage => {
    window.scrollTo(0, 0);
    history.push(`/?page=${newPage}`);
  };

  return (
    <>
      <Section>
        <SearchBox
          searchTerm={searchTerm}
          placeHolder="Search for movies"
          onSearchTermChanged={e => setSearchTerm(e.target.value)}
        />
      </Section>
      <Section>
        <Row align="right">
          <Pagination currentPage={currentPage} onPageChanged={pageChanged} totalPages={totalPages} />
        </Row>
        <MovieList movies={movies} empty={<Paragraph bold>No matches have been found.</Paragraph>} />
        <Row align="right">
          <Pagination currentPage={currentPage} onPageChanged={pageChanged} totalPages={totalPages} />
        </Row>
      </Section>
    </>
  );
};

export default Home;
