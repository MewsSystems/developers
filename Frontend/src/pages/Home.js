import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import Search from '../components/movies/Search';
import Movies from '../components/movies/Movies';
import Pagination from '../components/movies/Pagination';
import { searchMovies, setSearchTerm } from '../actions/movieActions';

const Home = ({ searchMovies, setSearchTerm, movie: { searchTerm } }) => {

  const handleSearchChange = query => {
    setSearchTerm(query);
    searchMovies(query);
  };

  const handleClick = number => {
    searchMovies(searchTerm, number);
  };

  return (
    <Fragment>
      <div className="container">
        <Search change={handleSearchChange} />
        <Movies />
        <Pagination click={handleClick} />
      </div>
    </Fragment>
  )
};

const mapStateToProps = state => ({
  movie: state.movie
});

export default connect(mapStateToProps, { searchMovies, setSearchTerm })(Home);
