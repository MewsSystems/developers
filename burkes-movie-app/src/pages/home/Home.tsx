import { debounce } from 'lodash';
import { useState } from 'react';
import { Link } from 'react-router-dom';

import { MovieCard } from '@/components/movieCard/MovieCard';
import { PageContainer } from '@/components/page/Page';
import { Pagination } from '@/components/pagination/Pagination';
import { useMoviesSearchQuery } from '@/queries/moviesQueries';

import css from './home.module.css';

export const Home = () => {
  const [search, setSearch] = useState('');
  const [currentPage, setCurrentPage] = useState(1);

  const handleSearch = (e: React.ChangeEvent<HTMLInputElement>) => {
    setSearch(e.target.value);
    setCurrentPage(1);
  };

  const debouncedSearch = debounce(handleSearch, 400);

  const {
    data,
    isLoading: isMovieListLoading,
    error,
  } = useMoviesSearchQuery(search, currentPage);

  const isSearchEmpty = !search;
  const isError = search && error;
  const isLoading = search && isMovieListLoading;
  const isFoundData = search && data?.results && data.results.length > 0;
  const isNoResults = search && data?.results && data.results.length === 0;

  return (
    <PageContainer>
      <div className={css.contentContainer}>
        <h1 className={css.title}>Mews Movie Search</h1>
        <input onChange={debouncedSearch} />
        <hr className={css.divider} />

        {isSearchEmpty && (
          <h1 className={css.title}>Please type something to begin search</h1>
        )}

        {isLoading && <h1 className={css.title}>Loading...</h1>}

        {isError && (
          <h1 className={css.title}>
            There seems to have been an error, the API might be down. Refer to
            the console for more details.
          </h1>
        )}

        {isFoundData && (
          <>
            <Pagination
              numberOfPages={data.total_pages}
              currentPage={currentPage}
              setCurrentPage={setCurrentPage}
            />

            <div className={css.movieCardsContainer}>
              {!isMovieListLoading && data && (
                <>
                  {data.results.map((movie) => (
                    <Link key={movie.id} to={`/${movie.id}`} state={movie}>
                      <MovieCard movie={movie} />
                    </Link>
                  ))}
                </>
              )}
            </div>
          </>
        )}

        {isNoResults && (
          <h1 className={css.title}>
            No results found, please alter your search
          </h1>
        )}
      </div>
    </PageContainer>
  );
};
