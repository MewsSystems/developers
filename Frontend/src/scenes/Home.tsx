import React, { useEffect, useState, useCallback } from 'react'
import { RouteComponentProps, navigate } from '@reach/router'
import { SceneMetaProps, SceneMeta } from 'components/SceneMeta'
import { Heading } from 'components/Heading'
import { useTranslation } from 'react-i18next'
import { useThunkDispatch } from 'hooks/useThunkDispatch'
import { hydrateTrendingMovies } from 'state/actions/movies'
import { useSelector } from 'react-redux'
import { State } from 'state/rootReducer'
import { differenceInDays } from 'date-fns'
import styled from 'styled-components'
import { MovieCard } from 'components/MovieCard'
import { Movie } from 'model/api/Movie'
import { Button } from 'components/Button'
import { useGetImagePath } from 'hooks/useGetImagePath'
import { ConfigurationBackdropSizesEnum } from 'model/api/ConfigurationBackdropSizesEnum'
import { Card as LoadingCard } from 'components/Loading/Card'
import { Spacing } from 'components/Spacing'
import { Content } from 'components/Content'

const Grid = styled.div`
  display: grid;
  grid-gap: 1rem;
  grid-template-columns: 1fr 1fr 1fr;
`

const { PUBLIC_URL } = process.env

export const Home: React.FC<RouteComponentProps<SceneMetaProps>> = ({
  title,
  description,
}) => {
  const { t } = useTranslation('home')
  const dispatch = useThunkDispatch()
  const [trendingMovies, setTrendingMovies] = useState<Movie[]>([])
  const [page, setPage] = useState(1)
  const {
    updatedAt,
    trending: { results, total_pages },
    loading,
  } = useSelector((state: State) => state.movies)
  const getImagePath = useGetImagePath(ConfigurationBackdropSizesEnum.W_780)
  const dayDifference = differenceInDays(new Date(), updatedAt)

  const handleLoadMore = useCallback(() => {
    setPage(page + 1)
  }, [page])

  const reduceTrendingMovies = useCallback(() => {
    return Object.keys(results).reduce<Movie[]>((acc, index) => {
      if (Number(index) <= page) {
        acc = [...acc, ...results[index]]
      }

      return acc
    }, [])
  }, [results, page])

  const handleCardClick = useCallback((id: number) => {
    navigate(`${PUBLIC_URL}/movie/${id}`)
  }, [])

  useEffect(() => {
    if (dayDifference > 7) {
      dispatch(hydrateTrendingMovies(true))
    } else {
      dispatch(hydrateTrendingMovies(false, page))
    }
  }, [dispatch, dayDifference, page])

  useEffect(() => {
    setTrendingMovies(reduceTrendingMovies())
  }, [page, results, reduceTrendingMovies])

  return (
    <>
      <SceneMeta title={title} description={description} />
      <Content>
        <Heading level={1}>{t('trending')}</Heading>
        {trendingMovies.length > 0 && (
          <>
            <Grid>
              {trendingMovies.map(
                ({ title, overview, backdrop_path, original_language, id }) => (
                  <MovieCard
                    key={id}
                    id={id}
                    title={title}
                    overview={overview}
                    background={getImagePath(backdrop_path)}
                    language={original_language}
                    onClick={handleCardClick}
                  />
                )
              )}
              {loading && (
                <>
                  <LoadingCard />
                  <LoadingCard />
                  <LoadingCard />
                  <LoadingCard />
                  <LoadingCard />
                </>
              )}
            </Grid>
            <Spacing outer="2rem 0 0">
              <Button
                onClick={handleLoadMore}
                disabled={page === total_pages || loading}
                fullWidth
              >
                {t('load-more')}
              </Button>
            </Spacing>
          </>
        )}
      </Content>
    </>
  )
}
