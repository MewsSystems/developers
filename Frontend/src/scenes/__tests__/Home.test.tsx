import React from 'react'
import { shallowWithStore } from 'test/shallowWithStore'
import { Home } from '../Home'

describe('Test Home scene', () => {
  it('Tests snapshot', () => {
    expect(shallowWithStore(<Home />)).toMatchSnapshot()
  })
})
