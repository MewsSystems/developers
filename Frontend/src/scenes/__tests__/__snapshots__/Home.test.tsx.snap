// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home scene Tests snapshot 1`] = `
<Fragment>
  <SceneMeta />
  <styled.div>
    <Heading
      level={1}
    >
      trending
    </Heading>
  </styled.div>
</Fragment>
`;
